docs/usage/DirectMigrationExampleAPI.md

```diff
@@ -0,0 +1,200 @@
+``` 
+oc create -f source-migcluster.yaml
+```
+
+**Note:** Please ensure that the `exposedRegistryPath` is appropriately specified, it is a must for
+facilitating **Direct Image Migration.**
+
+Before moving ahead, please verify that both the MigCluster instances are in a `Ready` state and there are no 
+critical conditions associated with these instances, if there are any issues or critical conditions present, please 
+resolve them.
+
+**2. Configure Migration storage**<br>
+This step involves the configuration of the object storage to be utilized during the 
+migration.
+
+For our example we will be configuring AWS S3 object storage as our MigStorage.
+
+Store the MigStorage yaml manifest in `migstorage.yaml`
+
+``` 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigStorage
+metadata:
+  name: aws-s3
+  namespace: openshift-migration
+spec:
+  backupStorageConfig:
+    awsBucketName: mybucket-6109f5e9711c8c58131acdd2f490f451
+    credsSecretRef:
+      name: aws-s3-jg74j
+      namespace: openshift-config
+  backupStorageProvider: aws
+  volumeSnapshotConfig:
+    credsSecretRef:
+      name: aws-s3-jg74j
+      namespace: openshift-config
+  volumeSnapshotProvider: aws
+```
+
+Command to create MigStorage instance:
+``` 
+oc create -f migstorage.yaml
+```
+
+**Note:** Please ensure that the secrets referenced in the `credsSecretRef` of 
+`backupStorageConfig` as well as the `volumeSnapshotConfig:` exist in the specifications provided.
+
+Once again, before moving further please ensure that the MigStorage instance is in `Ready` state 
+and there are not critical conditions associated with the same, if there are any issues or critical conditions present,
+please resolve them.
+
+**3. Create Migration Plan**<br>
+Now, switch the context to host cluster (where the Migration controller is deployed), let's create an
+instance of MigPlan.
+
+The yaml manifest of MigPlan consists of details pertaining to cluster references, storage reference,
+namespace reference of the application to be migration as well as the persistent volumes
+associated with the application to be migrated.
+
+Store the MigPlan yaml manifest in `migplan.yaml`
+
+``` 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigPlan
+metadata:
+  name: mtc-api-example
+  namespace: openshift-migration
+spec:
+  destMigClusterRef:
+    name: host
+    namespace: openshift-migration
+  indirectImageMigration: false
+  indirectVolumeMigration: false
+  migStorageRef:
+    name: aws-s3
+    namespace: openshift-migration
+  namespaces:
+    - django-app
+  persistentVolumes:
+    - capacity: 1Gi
+      name: pvc-336ecd0a-5ae1-11eb-aceb-029d2246b63f
+      pvc:
+        accessModes:
+          - ReadWriteOnce
+        hasReference: true
+        name: postgresql
+        namespace: django-app
+      selection:
+        action: copy
+        copyMethod: filesystem
+        storageClass: gp2
+        verify: false
+      storageClass: glusterfs-storage
```

Comment:
What is the difference between the `storageClass` in line 144 and the `storageClass`  in line 146?

Reply:
Another good thing to point out.

When you create your MigPlan initially, you're effectively telling the controller: "this is my intended migration, please execute discovery to tell me what you find in my source namespaces". The `persistentVolumes` array is written back to the MigPlan object with this structure. The top level `storageClass` field says "this is the source storage class". The `selection.storageClass` field is the desired target storage class. Really the entire `selection` object is the user's designation for how they want to process the associated PV.